app/models/mixins/cloud_tenancy_mixin.rb

```diff
@@ -0,0 +1,19 @@
+module CloudTenancyMixin
+  extend ActiveSupport::Concern
+
+  module ClassMethods
+    include TenancyCommonMixin
+
+    def tenant_id_clause_format(tenant_ids)
+      ["(tenants.id IN (?) AND ext_management_systems.tenant_mapping_enabled IS TRUE) OR ext_management_systems.tenant_mapping_enabled IS FALSE", tenant_ids]
```

Comment:
we have default value for ext_management_systems.tenant_mapping_enabled NULL
so can be included  also NULL  here as well as you have FALSE ?